src/redux/sagas/actions/addVerifiedMembers.ts

```diff
@@ -1,27 +1,148 @@
 
-function* addVerifiedMembers({
-  /*
+import {
+  createTransaction,
+  createTransactionChannels,
+  getTxChannel,
+} from '../transactions';
+import {
+  createActionMetadataInDB,
+  initiateTransaction,
+  putError,
+  takeFrom,
+  uploadAnnotation,
+} from '../utils';
+import { getAddVerifiedMembersOperation } from '../utils/verifiedMembers';
+
+function* addVerifiedMembersAction({
   payload: {
     colonyAddress,
     colonyName,
     members,
     customActionTitle,
     annotationMessage,
   },
-    */
-  // meta: { id: metaId, navigate, setTxHash },
+  meta: { id: metaId, navigate, setTxHash },
   meta,
 }: Action<ActionTypes.ACTION_ADD_VERIFIED_MEMBERS>) {
-  yield put<AllActions>({
-    type: ActionTypes.ACTION_ADD_VERIFIED_MEMBERS_SUCCESS,
-    payload: {},
-    meta,
-  });
+  let txChannel;
+
+  try {
+    if (!colonyAddress) {
+      throw new Error(
+        'No colony address set for addVerifiedMembers transaction',
+      );
+    }
+    if (!Array.isArray(members) || members.length === 0) {
+      throw new Error('No members set for addVerifiedMembers transaction');
+    }
+
+    txChannel = yield call(getTxChannel, metaId);
+
+    // setup batch ids and channels
+    const batchKey = 'addVerifiedMembers';
+
+    const { addVerifiedMembers, annotateAddVerifiedMembers } =
+      yield createTransactionChannels(metaId, [
+        'addVerifiedMembers',
+        'annotateAddVerifiedMembers',
+      ]);
+
+    yield fork(createTransaction, addVerifiedMembers.id, {
+      context: ClientType.ColonyClient,
+      methodName: 'editColonyByDelta',
+      identifier: colonyAddress,
+      params: [],
+      group: {
+        key: batchKey,
+        id: metaId,
+        index: 0,
+      },
+      ready: false,
+    });
+
+    if (annotationMessage) {
+      yield fork(createTransaction, annotateAddVerifiedMembers.id, {
+        context: ClientType.ColonyClient,
+        methodName: 'annotateTransaction',
+        identifier: colonyAddress,
+        params: [],
+        group: {
+          key: batchKey,
+          id: metaId,
+          index: 1,
+        },
+        ready: false,
+      });
+    }
+
+    yield takeFrom(addVerifiedMembers.channel, ActionTypes.TRANSACTION_CREATED);
+    if (annotationMessage) {
+      yield takeFrom(
+        annotateAddVerifiedMembers.channel,
+        ActionTypes.TRANSACTION_CREATED,
+      );
+    }
+
+    // add params, how to blobify the last JSON??
+    yield put(
+      transactionAddParams(addVerifiedMembers.id, [
+        JSON.stringify(getAddVerifiedMembersOperation(colonyAddress, members)),
+      ]),
+    );
+
+    yield initiateTransaction({ id: addVerifiedMembers.id });
+
+    const {
+      payload: { hash: txHash },
+    } = yield takeFrom(
+      addVerifiedMembers.channel,
+      ActionTypes.TRANSACTION_HASH_RECEIVED,
+    );
+
+    setTxHash?.(txHash);
+
+    yield takeFrom(
+      addVerifiedMembers.channel,
+      ActionTypes.TRANSACTION_SUCCEEDED,
+    );
```

Comment:
I believe this should be replaced with `waitForTxResult` (#1018 currently in progress to replace these globally)